[WIP] KLayout 0.25: No.2 Compile Errors on Mac OSX

Facing another build issue...

MacBookPro{sekigawa}(250)$ /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ --version
Apple LLVM version 7.0.2 (clang-700.1.81)
Target: x86_64-apple-darwin14.5.0
Thread model: posix

----

/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -headerpad_max_install_names -stdlib=libc++ -Wl,-syslibroot,/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -mmacosx-version-min=10.9  -Wl,-rpath,@loader_path/./qt5.bin.macos-yosemite -single_module -dynamiclib -compatibility_version	0.25 -current_version	0.25.0 -install_name	libdb_tests.0.dylib -o libdb_tests.0.25.0.dylib dbArray.o dbBox.o dbBoxScanner.o dbBoxTree.o dbCell.o dbCellGraphUtils.o dbCellHullGenerator.o dbCellMapping.o dbCIFReader.o dbClip.o dbDXFReader.o dbExpression.o dbEdge.o dbEdgePair.o dbEdgePairRelations.o dbEdgePairs.o dbEdgeProcessor.o dbEdges.o dbEdgesToContours.o dbGDS2Reader.o dbGDS2Writer.o dbLayer.o dbLayerMapping.o dbLayout.o dbLayoutDiff.o dbLayoutUtils.o dbLayoutQuery.o dbLibraries.o dbMatrix.o dbOASISReader.o dbOASISWriter2.o dbOASISWriter.o dbObject.o dbPath.o dbPCells.o dbPoint.o dbPolygon.o dbPolygonTools.o dbPropertiesRepository.o dbRecursiveShapeIterator.o dbRegion.o dbShapeArray.o dbShape.o dbShapeRepository.o dbShapes.o dbStreamLayers.o dbText.o dbTilingProcessor.o dbTrans.o dbVector.o dbWriterTools.o dbVariableWidthPath.o  -F/opt/local/libexec/qt5/lib -L/Users/sekigawakazunari/GitWork/klayout-0.25/qt5.build.macos-yosemite/db/unit_tests/../.. -lklayout_db -lklayout_tl -lklayout_gsi -framework QtDesigner -framework QtWidgets -framework QtGui -framework QtCore -framework DiskArbitration -framework IOKit -framework QtXml -framework QtPrintSupport -framework QtMultimediaWidgets -framework QtMultimedia -framework QtNetwork -framework QtSvg -framework QtXmlPatterns -framework QtSql -framework QtTest -framework Security -framework Foundation -framework ApplicationServices -framework OpenGL -framework AGL  
Undefined symbols for architecture x86_64:
  "vtable for db::GDS2Writer", referenced from:
      run_test(tl::TestBase*, char const*, char const*, char const*, double, bool, bool) in dbCIFReader.o
      write(std::__1::vector<db::polygon<int>, std::__1::allocator<db::polygon<int> > > const&, std::__1::vector<db::polygon<int>, std::__1::allocator<db::polygon<int> > > const&, std::__1::vector<db::edge<int>, std::__1::allocator<db::edge<int> > > const&, std::__1::vector<db::edge<int>, std::__1::allocator<db::edge<int> > > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in dbEdgeProcessor.o
      run_test(tl::TestBase*, char const*, char const*) in dbGDS2Writer.o
      (anonymous namespace)::TestImpl2::execute(tl::TestBase*) in dbGDS2Writer.o
      (anonymous namespace)::TestImpl3::execute(tl::TestBase*) in dbGDS2Writer.o
      (anonymous namespace)::TestImpl4::execute(tl::TestBase*) in dbGDS2Writer.o
      run_test(tl::TestBase*, char const*, bool, int, bool) in dbOASISWriter.o
      ...
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [../../db/libdb_tests.0.25.0.dylib] Error 1
make[1]: *** [sub-unit_tests-make_first] Error 2
make: *** [sub-db-make_first] Error 2

----
Other companion tools were normally built but do not work (including klayout.app) --> #6.

MacBookPro{sekigawa}(191)$ ll | grep str
drwxr-xr-x   3 sekigawa staff       102 12 12 23:10 strm2cif.app
drwxr-xr-x   3 sekigawa staff       102 12 12 23:10 strm2dxf.app
drwxr-xr-x   3 sekigawa staff       102 12 12 23:10 strm2gds.app
drwxr-xr-x   3 sekigawa staff       102 12 12 23:10 strm2gdstxt.app
drwxr-xr-x   3 sekigawa staff       102 12 12 23:10 strm2oas.app
drwxr-xr-x   3 sekigawa staff       102 12 12 23:10 strm2txt.app
drwxr-xr-x   3 sekigawa staff       102 12 12 23:10 strmclip.app
drwxr-xr-x   3 sekigawa staff       102 12 12 23:10 strmcmp.app
drwxr-xr-x   3 sekigawa staff       102 12 12 23:10 strmrun.app
drwxr-xr-x   3 sekigawa staff       102 12 12 23:10 strmxor.app

--- a/src/db/db/dbGDS2Writer.cc
+++ b/src/db/db/dbGDS2Writer.cc
@@ -45,34 +45,34 @@ GDS2Writer::GDS2Writer ()
   m_progress.set_unit (1024 * 1024);
 }
 
-inline void 
+void 
 GDS2Writer::write_byte (unsigned char b)
 {
   mp_stream->put ((const char *) &b, 1);
 }
 
-inline void 
+void 
 GDS2Writer::write_record_size (int16_t i)
 {
   gds2h (i);
   mp_stream->put ( (char*)(&i), sizeof (i));
 }
 
-inline void 
+void 
 GDS2Writer::write_record (int16_t i)
 {
   gds2h (i);
   mp_stream->put ( (char*)(&i), sizeof (i));
 }
 
-inline void 
+void 
 GDS2Writer::write_short (int16_t i)
 {
   gds2h (i);
   mp_stream->put ( (char*)(&i), sizeof (i));
 }
 
-inline void 
+void 
 GDS2Writer::write_int (int32_t l)
 {
   gds2h (l);

